wandb/sdk/wandb_run.py

```diff
@@ -3701,6 +3705,65 @@ def pb_to_dict(
                 logger.error("Error getting system metrics: %s", e)
         return {}
 
+    @property
+    @_run_decorator._attach
+    @_run_decorator._noop_on_finish()
+    def _metadata(self) -> Metadata | None:
+        """The metadata associated with this run.
+
+        NOTE: Automatically collected metadata can be overridden by the user.
+        """
+        if self.settings.x_require_legacy_service:
+            wandb.termwarn(
+                "The metadata property is not supported when using the legacy service."
+            )
+            return None
+
+        if not self._backend or not self._backend.interface:
+            return self.__metadata
+
+        # Initialize the metadata object if it doesn't exist.
+        if self.__metadata is None:
+            self.__metadata = Metadata()
+            self.__metadata._set_callback(self._metadata_callback)
+
+        handle = self._backend.interface.deliver_get_system_metadata()
```

Comment:
so everytime a user access this object we are going to to do a getter call?

Reply:
Yes, that's the idea, bc the object is updated by core + the user can also modify it.

Reply:
oh we don't have guarantee that it won't change by another process ...

Reply:
in that case (if it's not attacdh), the last update wins, which is fair, I think.